src/StockTransactionsInterface.php

```diff
@@ -33,7 +47,7 @@
    * @return \Drupal\commerce_stock\StockLocationInterface
    *   The stock location.
    */
-  public function getPrimaryTransactionLocation(PurchasableEntityInterface $entity, $quantity);
+  public function getTransactionLocation(Context $context, PurchasableEntityInterface $entity, $quantity);
```

Comment:
The stock transaction manager, whatever that may be (definitely shouldn't be the stock service manager), shouldn't be handling location determination ... it should rely on a location management/resolution service.